src/tools/genpolicy/src/policy.rs

```diff
@@ -1092,6 +1107,18 @@ fn get_container_annotations(
         annotations
             .entry("io.kubernetes.cri.image-name".to_string())
             .or_insert(image_name);
+
+        // Add CDI annotation regex pattern for GPU containers
+        // The actual device numbers are assigned at runtime by the shim
+        // Pattern matches: cdi.k8s.io/vfio<num> = nvidia.com/gpu=<num>
+        if let Some(nvidia_pgpu_count) = yaml_container.get_nvidia_pgpu_count() {
+            if nvidia_pgpu_count > 0 {
+                annotations.insert(
+                    "^cdi\\.k8s\\.io/vfio[0-9]+$".to_string(),
+                    "^nvidia\\.com/gpu=[0-9]+$".to_string(),
```

Comment:
"^nvidia\\.com/gpu=[0-9]+$" should be read by genpolicy from genpolicy-settings.json - that way hopefully supporting other vendor regex strings in the future too. An example of handling regex strings through genpolicy-settings.json:

            "forbidden_source_regex": [
                "^(?:0{0,4}:){0,7}0{0,3}1$",
                "^127\\.(?:[0-9]{1,3}\\.){2}[0-9]{1,3}$"
            ]